app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -88,18 +37,15 @@ fun HomeRoute(
                 MovieCategory.POPULAR_RENT.ordinal,
                 MovieCategory.POPULAR_ONTV.ordinal,
                 MovieCategory.POPULAR_THEATERS.ordinal -> {
-                    popularCategoryViewState =
-                        changeMovieFavoriteStatus(popularCategoryViewState, movie)
+                    homeViewModel.toggleFavorite(movie.id)
                 }
                 MovieCategory.PLAYING_MOVIES.ordinal,
                 MovieCategory.PLAYING_TV.ordinal -> {
-                    nowPlayingCategoryViewState =
-                        changeMovieFavoriteStatus(nowPlayingCategoryViewState, movie)
+                        homeViewModel.toggleFavorite(movie.id)
                 }
                 MovieCategory.UPCOMING_TODAY.ordinal,
                 MovieCategory.UPCOMING_WEEK.ordinal -> {
-                    upcomingCategoryViewState =
-                        changeMovieFavoriteStatus(upcomingCategoryViewState, movie)
+                        homeViewModel.toggleFavorite(movie.id)
```

Comment:
you are calling the same function regardless of movie category, so why don't you call it just once? 

```kotlin
onFavoriteClick = { _, movie ->
    homeViewModel.toggleFavorite(movie.id)
}
```